recovery_behaviour_enabled: true
 
GlobalPlanner:
  allow_unknown: true # Specifies whether or not to allow navfn to create plans that traverse unknown space.
  default_tolerance: 0.1 # A tolerance on the goal point for the planner.
  visualize_potential: false
  use_dijkstra: true
  use_quadratic: true
  use_grid_path: false
  old_navfn_behavior: false
  lethal_cost: 100
  neutral_cost: 75
  cost_factor: 3.0
  publish_potential: true
  orientation_mode: 0
  orientation_window_size: 1
  outline_map: false

DWAPlannerROS:
  controller_frequency: 20

  # Robot configuration parameters  
  acc_lim_x: 1.0 #0.1
  acc_lim_y: 0.0 #0.1
  acc_lim_th: 2.0 #0.1

  max_vel_x: 0.07 #0.5
  min_vel_x: -0.03 #0.1
  max_vel_y: 0
  min_vel_y: 0

  max_vel_trans: 0.07
  min_vel_trans: 0.01
  max_vel_theta: 0.1
  min_vel_theta: 0.05

  # Goal Tolerance Parameters
  yaw_goal_tolerance: 0.5 #0.4
  xy_goal_tolerance: 0.1
  latch_xy_goal_tolerance: false

  # Forward Simulation Parameters
  sim_time: 1.7
  sim_granularity: 0.01
  vx_samples: 20
  vy_samples: 0
  vth_samples: 20

  # # Trajectory scoring parameters
  path_distance_bias: 32.0 # The weighting for how much the controller should stay close to the path it was given
  goal_distance_bias: 24.0 # The weighting for how much the controller should attempt to reach its local goal, also controls speed
  occdist_scale: 0.01 # The weighting for how much the controller should attempt to avoid obstacles
  forward_point_distance: 0.325 # The distance from the center point of the robot to place an additional scoring point, in meters
  stop_time_buffer: 0.5  # The amount of time that the robot must stThe absolute value of the veolicty at which to start scaling the robot's footprint, in m/sop before a collision in order for a trajectory to be considered valid in seconds
  scaling_speed: 0.25 # The absolute value of the veolicty at which to start scaling the robot's footprint, in m/s
  max_scaling_factor: 0.2 # The maximum factor to scale the robot's footprint by
  prune_plan: true
  # # Oscillation Prevention Parameters
  # oscillation_reset_dist: 0.25 #How far the robot must travel in meters before oscillation flags are reset (double, default: 0.1)
